pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/DefaultIndexCreatorProvider.java

```diff
@@ -76,6 +75,10 @@ public ForwardIndexCreator newForwardIndexCreator(IndexCreationContext.Forward c
       throws Exception {
     if (!context.hasDictionary()) {
       // Dictionary disabled columns
+      if (context.forwardIndexDisabled()) {
```

Comment:
(minor) Suggest moving the check to the caller side to be consistent with other index creators. Basically when invoking this method, forward index should be created

Reply:
done